python/ray/dag/compiled_dag_node.py

```diff
@@ -867,6 +878,76 @@ def _get_node_id(self):
 
         self._dag_submitter.start()
         self._dag_output_fetcher.start()
+        self._capacity = self._compute_capacity()
+        print(f"Capacity of the DAG: {self._capacity}")
+
+    @property
+    def capacity(self) -> int:
+        return self._capacity
+
+    def _compute_capacity(self):
+        """
+        Compute the capacity of the DAG. The capacity is the number of inputs
+        the DAG could start processing without blocking when none of the outputs
+        is consumed.
+
+        Returns:
+            The capacity of the DAG.
+
+        NOTE: The DAG can hold intermediate results in either the intermediate
+        channels or the actors, specifically:
+        - Channel capacity: either 1 or 0 depending on the channel type.
+        - Actor capacity: capacity is 1 because the actor can fetch and hold one
+            object from each input channel during execution. For an actor with
+            multiple methods, the capacity is still 1 because the current method
+            needs to finish (which flushes the held object) before the next method
+            can start.
+
+        This method performs a modified Dijkstra's algorithm to compute the
+        capacity of the DAG.
+        """
+        capa = {idx: float("inf") for idx in self.idx_to_task}
+        pq = []
+
+        # Initial capacity is 1 as input channel has capacity 1.
+        capa[self.input_task_idx] = 1
+        heapq.heappush(pq, (1, self.input_task_idx, set()))
+
+        while pq:
+            cur_capa, cur_idx, cur_actors = heapq.heappop(pq)
+            if cur_idx == self.output_task_idx:
```

Comment:
Add a comment to explain why output channel capacity is not considered.

Reply:
Added a comment: it is considered together with the actor "node".